### PR TITLE
Frenzy immunities fixed, less broken genies spellcasting

### PR DESCRIPTION
**Pull request by dydzio0614**
Frenzy immunities are closer to H3 (orb of vulnerability special behavior not handled)

Also a bit better genies spellcasting check until i resigned from continuing

### CODE REVIEW

File: lib/battle/CBattleInfoCallback.cpp [hunk 1]
```diff
@@ -1673,7 +1674,7 @@ SpellID CBattleInfoCallback::getRandomBeneficialSpell(CRandomGenerator & rand, c
 				return !stack->canShoot();
 			});
 
-			if(!walker)
+			if(!walker || subject->hasBonusOfType(BonusType::SPELL_SCHOOL_IMMUNITY, SpellSchool::FIRE))
```
Comment (IvanSavenko) [inline lib/battle/CBattleInfoCallback.cpp:1677]: This block is also used by Shield spell, which is earth spell

Comment (IvanSavenko) [reply lib/battle/CBattleInfoCallback.cpp:1677]: Strange. Why this block is needed? Code actually checks whether spell can be casted on this target - see `canBeCast` block above.

Comment (dydzio0614) [reply lib/battle/CBattleInfoCallback.cpp:1677]: the block does not work - if you try to cast genie spell on ballista then half of the attempts fail

Comment (IvanSavenko) [reply lib/battle/CBattleInfoCallback.cpp:1677]: Ah, we actually check whether spell can be cast *at all*, not whether spell can be cast on this target. Will fix.

**Comment by IvanSavenko** [comment]
Closing, included in #3379

**Pull request closed without merging**